modules/zio2/testkit/src/main/scala/com/caesars/tracing/TracingTestUtils.scala

```diff
@@ -104,7 +104,7 @@ object ZTracerImplementationSpecUtils {
 
     for {
       tracedApp1 <- TracedHttp.layer()(app)
-      _          <- tracedApp1(Request(method = Method.GET, url = URL(path = Path(Vector("foo"), trailingSlash = false)))).orDieWith(_.get)
+      _          <- tracedApp1(Request(url = URL.fromString("/foo").getOrElse(???))).either
```

Comment:
Is this an omission?

Reply:
nope, just simplifying

Reply:
Then instead, why not storing that in a variable, or even better refactor the tests using a function:
```scala
def mkApp(app: ...) = {
    for {
      tracedApp <- TracedHttp.layer()(app)
      _         <- tracedApp(Request(method = Method.GET, url = URL(path = Path(Vector("foo"), trailingSlash = false)))).either
      ref       <- ZIO.service[SpanRecorder]
      spans     <- ref.get
      span = spans.head
    } yield span
}
```
and then reuse this function everywhere. This simplifies the code and get rid of some boilerplate. Personally, I'm not a big fan of keeping `???` in the code. This sounds like something that is not implemented. Secondly, I'm not sure about what's the type of `URL.fromString`, but it it's an option, then one might better call `get`.

If none of this is possible, then I would simply store the request in a value:
```scala
val get = Request(method = Method.GET, url = URL(path = Path(Vector("foo"), trailingSlash = false)))
```

Reply:
That is valid and needed, but not what I came into this codebase to do ;)
Lets see if I can improve this...